### ISSUE TITLE
Subdirectory on a Server Object’s url field is getting cut off

### ISSUE THREAD

**Issue by jimmymcpeter** [mswjs/source#58]
## Description

The subdirectory on a Server Object’s url field is getting cut off.

[from-open-api.ts#L55-L57](https://github.com/mswjs/source/blob/main/src/open-api/from-open-api.ts#L55-L57)
```ts
const requestUrl = isAbsoluteUrl(baseUrl)
                    ? new URL(path, baseUrl).href
                    : joinPaths(path, baseUrl);
```

## Reproduction steps

api-schema.json
```json
"servers": [
    {
      "url": "https://example.com/web"
    }
  ],
  "paths": {
    "/api/v1/whoAmI": {
      "get": {
        "summary": "Who am I",
        "description": "",
        "operationId": "whoAmI",
        "responses": {
          "200": {
            ...
          }
        }
      }
    }
  },
```

The requestUrl const = `https://example.com/api/v1/whoAmI`

## Expected behavior

The requestUrl const should = `https://example.com/web/api/v1/whoAmI`

## Initial assessment

The path should be appended to the Server Object’s url field in order to construct the full URL.  Calling `new URL()` is how it's getting cut off.

## Screenshots

![image](https://github.com/user-attachments/assets/5215998e-9bc1-4d6e-a11a-380b7ca7ce43)

**Comment by weyert** [comment]
Ah interesting. We always the full path in the operations instead in servers :)

**Comment by kettanaito** [comment]
Thanks for reporting this, @jimmymcpeter! I've reviewed your PR, I think we're almost there.

**Comment by kettanaito** [comment]
## Released: v0.2.1 🎉

This has been released in v0.2.1!

- 📄 [**Release notes**](https://github.com/mswjs/source/releases/tag/v0.2.1)
- 📦 [npm package](https://www.npmjs.com/package/@mswjs/source/v/0.2.1)

Make sure to always update to the latest version (`npm i @mswjs/source@latest`) to get the newest features and bug fixes.

---

_Predictable release automation by [@ossjs/release](https://github.com/ossjs/release)_.
